spec/faraday/utils/headers_spec.rb

```diff
@@ -56,6 +56,15 @@
     it { expect(subject.delete('content-type')).to be_nil }
   end
 
+  describe '#dig' do
+    before { subject['Content-Type'] = 'application/json' }
+
+    it { expect(subject&.dig('Content-Type')).to eq('application/json') }
+    it { expect(subject&.dig('CONTENT-TYPE')).to eq('application/json') }
+    it { expect(subject&.dig(:content_type)).to eq('application/json') }
+    it { expect(subject&.dig('invalid')).to be_nil }
```

Comment:
The safe-navigation operator is not needed here.

(The use of the `subject` dynamic definition's superfluous, too, but make no changes to that in this PR.)